src/approximate_inference.jl

```diff
@@ -0,0 +1,32 @@
+import Base: rand, length
+import Distributions: logpdf, AbstractMvNormal
+
+export elbo, dtc
+export SparseFiniteGP
+
+struct SparseFiniteGP{T1<:FiniteGP, T2<:FiniteGP} <: AbstractMvNormal
+    fobs::T1
+    finducing::T2
+end
+
+length(f::SparseFiniteGP) = length(f.fobs)
+mean(f::SparseFiniteGP) = mean(f.fobs)
+covariance_error = "The covariance matrix of a sparse GP can often be dense and " *
+    "can cause the computer to run out of memory. If you are sure you have enough " *
+    "memory, you can use `cov(f.fobs)`."
+cov(f::SparseFiniteGP) = error(covariance_error) #cov(f.fobs)
+# Not sure how to implement the following...
+# cov(fx::FiniteGP, gx::FiniteGP) = cov(fx.f, gx.f, fx.x, gx.x)
+marginals(f::SparseFiniteGP) = Normal.(mean(f), sqrt.(cov_diag(f.fobs.f, f.fobs.x) .+ diag(f.fobs.Σy)))
```

Comment:
I wonder if this could just call `marginals(f.fobs)`, instead of being written out in full?